validation-test/stdlib/SequenceType.swift.gyb

```diff
@@ -982,6 +982,11 @@ SequenceTypeTests.test("prefix/dispatch") {
   expectCustomizable(tester, tester.log.prefixMaxLength)
 }
 
+SequenceTypeTests.test("prefix/drop/dispatch") {
+  let xs = sequence(first: 1, next: {$0 < 10 ? $0 + 1 : nil})
+  expectTrue(Array(xs.prefix(3).drop(while: {$0 < 7})).isEmpty)
```

Comment:
I can prepare a patch for master that does this.  Do you want this to also land with this PR in 3.1 or does this fix suffice on its own?